chirp/drivers/kguv980p.py

```diff
@@ -0,0 +1,2583 @@
+        except errors.RadioError:
+            raise
+        except Exception, e:
+            LOG.exception('Unknown error during download process')
+            raise errors.RadioError("Failed to communicate with radio: %s" % e)
+
+    def _do_download(self, start, end, blocksize):
+        # allocate & fill memory
+        LOG.debug("Start Download")
+        image = ""
+        for i in range(start, end, blocksize):
+            req = chr(i / 256) + chr(i % 256) + chr(blocksize)
+            self._write_record(CMD_RD, req)
+            cs_error, resp = self._read_record()
+            LOG.debug("Got:\n%s" % util.hexprint(resp))
+
+            if cs_error:
+                LOG.debug(util.hexprint(resp))
+                raise Exception("Checksum error on read")
+            # LOG.debug("Got:\n%s" % util.hexprint(resp))
+            image += resp[2:]
+            if self.status_fn:
+                status = chirp_common.Status()
+                status.cur = i
+                status.max = end
+                status.msg = "Cloning from radio"
+                self.status_fn(status)
+        self._finish()
+        return memmap.MemoryMap(''.join(image))
+
+    def _upload(self):
+        """Talk to a wouxun KG-UV980P and do a upload"""
+        try:
+            self._identify()
+            LOG.debug("Done with Upload Identify")
+            # self._do_upload(0, 1856, 16)
+            # LOG.debug("Done with Limits Upload")
+            # self._do_upload(1856, 32768, 64)
+            self._do_upload()
+            LOG.debug("Done with Mem and Settings Upload")
+            self._finish()
+        except errors.RadioError:
+            raise
+        except Exception, e:
+            raise errors.RadioError("Failed to communicate with radio: %s" % e)
+        return
+
+    def _do_upload(self):
+        LOG.debug("Start of _do_upload")
+        for ar, size, count in config_map:
+            for addr in range(ar, ar + (size*count), size):
+                start = ar
+                ptr = start
+                LOG.debug("ptr = " + str(ptr))
+                end = ar + (size*count)
+                blocksize = size
+                LOG.debug("Start of loop in _do_upload index = "+str(addr))
+                LOG.debug("Start %i End %i Size %i" % (start, end, blocksize))
+                req = chr(addr / 256) + chr(addr % 256)
+                LOG.debug("REQ")
+                chunk = self.get_mmap()[addr:addr + blocksize]
+                LOG.debug("CHUNK")
+                self._write_record(CMD_WR, req + chunk)
+    #            LOG.debug("Upload-- SENT : " % util.hexprint(_sent))
+                cserr, ack = self._read_record()
+                LOG.debug("Upload-- CSUM ERROR : " + str(cserr))
+                LOG.debug("Upload-- RCVD :\n%s " % util.hexprint(ack))
+
+                j = ord(ack[0]) * 256 + ord(ack[1])
+                LOG.debug("j = " + str(j))
+                LOG.debug("addr= " + str(addr))
```

Comment:
This seems like overly verbose tracing that is probably not necessary to spam into everyone's logs every time, right? All the debug logging in this inner loop seems like it should be trimmed out after getting it working.

Reply:
agree - it could be.. but it does help especially if a user has a problem that I did not experience.

Reply:
Okay, but for every block you're logging 9 lines, some of which are just static strings. How about you collapse that down to one LOG.debug() call with the relevant bits?

Reply:
agree - I will remove the static logs and minimize or limit writes to the log to when an error is detected.